sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.cs

```diff
@@ -758,5 +758,20 @@ public int Start()
         }
 
         public void Archive() {}
+
+        public class TestClass
+        {
+            public int VarPatternCheck(string value)
+            {
+                switch (value)
+                {
+                    case var x when x == "one":

```

Comment:
Is this a regression test for the bug? I tried executing this test with the new `VarPatternSyntaxWrapper` class commented out and the DeadStores test didn't fail.

Reply:
Indeed it only fails when running with MSBuild 16.

The way I tested it was to create a small project with this class and run the analyzer on it:
"%MSBUILD16%" "ConsoleApp.sln" /t:Clean,Build

and then update the .csproj to use the local analyzer and run again
```
 <ItemGroup>
    <Analyzer Include="..\..\..\..\..\..\src\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\SonarAnalyzer.dll" />
    <Analyzer Include="..\..\..\..\..\..\src\sonar-dotnet\sonaranalyzer-dotnet\src\SonarAnalyzer.CSharp\bin\Debug\net46\SonarAnalyzer.CSharp.dll" />
  </ItemGroup>
```

The worst part now is that we get 0 coverage for the new code since it will never run with the current build due to the fact that we use MsBuild 15.

Reply:
Thanks for the explanation. As discussed offline, please create a ticket about also running tests using MSBuild16.